cricket_situations

ai-scale opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, __failure to follow the guidelines below will result in the PR being closed automatically__. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access granted. 🚨

__PLEASE READ THIS__:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

Also, pelase note that we're using **Git LFS** for storing the JSON files, so please make sure that you move the JSON file to Git LFS before submitting a PR. Details on how to use Git LFS are available [here](https://git-lfs.com).

## Eval details 📑
### Eval name
cricket_situations

### Eval description

It asks the model whether it is possible to win a cricket game from a given situation.

### What makes this a useful eval?

It tests whether the model is able to apply the rules of the game at various situations especially in edge cases.

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [x] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can  create an eval on cases where the model fails to reason about the physical world.
- [x] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [x] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [x] **Include at least 15 high quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

The accuracy we got on GPT 3.5 is 0%.

## Eval structure 🏗️

Your eval should
- [x] Check that your data is in `evals/registry/data/{name}`
- [x] Check that your yaml is registered at `evals/registry/evals/{name}.yaml`
- [x] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (https://platform.openai.com/docs/usage-policies).

- [x] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the merged pull request.

- [x] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgement

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [x] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access granted.

### Submit eval

- [x] I have filled out all required fields of this form
- [x] I have used **Git LFS** for the Eval JSON data
- [ ] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data 

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the batting team to win the match if they need 33 runs to win from 5 balls assuming that the batting team is extremely lucky"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the batting team to win the match if they need 34 runs to win from 5 balls assuming that the batting team is extremely lucky"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the batting team to win the match if they need 35 runs to win from 5 balls assuming that the batting team is extremely lucky"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the batting team to win the match if they need 36 runs to win from 5 balls assuming that the batting team is extremely lucky"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the batting team to win the match if they need 37 runs to win from 5 balls assuming that the batting team is extremely lucky"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the bowling team to win the match if the batting team needs 1 run to win from 1 ball, a tie is not considered as a win"}], "ideal": "NO"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the bowling team to win the match if the batting team needs 1 run to win from 2 balls, a tie is not considered as a win"}], "ideal": "NO"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the bowling team to win the match if the batting team needs 1 run to win from 3 balls, a tie is not considered as a win"}], "ideal": "NO"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the bowling team to win the match if the batting team needs 1 run to win from 4 balls, a tie is not considered as a win"}], "ideal": "NO"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the bowling team to win the match if the batting team needs 1 run to win from 5 balls, a tie is not considered as a win"}], "ideal": "NO"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the batting team to tie the match if they require 7 runs to win from 1 ball assuming all balls are legal"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the batting team to tie the match if they require 13 runs to win from 2 balls assuming all balls are legal"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the batting team to tie the match if they require 19 runs to win from 3 balls assuming all balls are legal"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the batting team to tie the match if they require 25 runs to win from 4 balls assuming all balls are legal"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible for the batting team to tie the match if they require 31 runs to win from 5 balls assuming all balls are legal"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Let's say my team needs 3 runs to win of 7 balls and I am on strike and this is the last wicket. I am currently batting on 92. Is it possible for me to score a 100?"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Let's say my team needs 9 runs to win of 8 balls and I am on strike and this is the last wicket. I am currently batting on 86. Is it possible for me to score a 100?"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Let's say my team needs 15 runs to win of 9 balls and I am on strike and this is the last wicket. I am currently batting on 80. Is it possible for me to score a 100?"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Let's say my team needs 21 runs to win of 10 balls and I am on strike and this is the last wicket. I am currently batting on 74. Is it possible for me to score a 100?"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Let's say my team needs 27 runs to win of 11 balls and I am on strike and this is the last wicket. I am currently batting on 68. Is it possible for me to score a 100?"}], "ideal": "YES"}
{"input": [{"role": "system", "content": "You will be given various situations in the game of cricket along with an outcome and you have to answer whether the outcome is possible with a YES or NO and no additional text"}, {"role": "user", "content": "Is it possible to score 5 runs on a legal delivery without the ball touching the bat"}], "ideal": "YES"}
  ```
</details>

usama-openai reviewed (changes requested):
Thanks for the contribution. I would like to request some changes.

1. The extension of the eval definition file is `.yaml`, but you have created the file with the extension `.yml`. Kindly fix this issue.

2. Add a proper description of the eval in the `.yaml` file.

3. The extension of the samples file is `.jsonl`, but you have created the file with the extension `.json`. Kindly fix this issue.

4. Revert all the changes in the `.gitattributes` file.

5. The naming convention for evals is in the form `<eval_name>.<split>.<version>`.
    - `<eval_name>` is the eval name, used to group evals whose scores are comparable.
    - `<split>` is the data split, used to further group evals that are under the same <base_eval>. E.g., val, test, or dev for testing.
    - `<version>` is the version of the eval, which can be any descriptive text you'd like to use (though it's best if it does not contain.).

    You can rename the eval id `cricket_situations.v0` to `cricket_situations.dev.v0`.

We would love to review the PR again after the suggested changes.

ai-scale commented:
@usama-openai 

Thanks for reviewing my PR.

I have made the requested changes.  I have changed to json file to jsonl and the pml file to paml. Please let me know if any other changes are to be made.

usama-openai reviewed (changes requested):
Thanks for implementing the requested changes. I would like to request a few more changes.

1. Kindly revert all the changes in the `.gitattributes` file. You don't need to make any changes to this file. If git lfs is properly installed, the required files will automatically be pushed as lfs files.

2. I have the following reservations about the dataset:
    - Currently, all the samples in the dataset are of the same type or situation. Try to think of various situations in the game.
    - All the samples have situations with the possibility of winning. Add samples with the situations where winning is not possible to make the dataset balanced.
    - All the situations ask about the winning possibility of the batting side. Add samples to ask for the winning possibility of the bowling side as well. It can be achieved by implementing the above two suggestions, which will also make the dataset diverse and balanced.
    - By looking at the ideal answer, it seems like some results highly depend on luck. For example, consider the following situation:

        `Is it possible for the batting team to win the match if they need 31 runs to win from 4 balls`

        The ideal answer is `Yes` which is logically correct, but the batting side has to be very lucky, and the batsman has to hit 5 sixes with one or more no or wide balls. If we go with this logic, then it is logically possible to win in the situation `if they need 300 runs to win from 1 ball`. So, if you want to use such situations, then I would advise you to add proper instructions in the prompt for the model, like `don't add a luck factor` or `assume that the batting/bowling side is very lucky`.

We would love to review the PR again after the suggested changes.

ai-scale commented:
@usama-openai 

Thanks for your feedback. 

1) I have reverted all the changes to the .gitattributes file.
2) I have added the sentence " assuming the batting team is extremely lucky to the prompts".
3) I have added examples where we are asking if it's possible for the bowling team to win.
4) I have also added additional situations where the model needs to consider the possibility of a "short run".

I have only given cases where the batting team requires some sort of luck to win because even though the model is aware about "wides" and "no balls", it does not consider them in these situations.

usama-openai commented:
Thanks for the updates. There is a small issue with the dataset. `Is it possible for the batting team to win the match if they need 37 runs to win from 5 balls, assuming that the batting team is extremely lucky` is repeated twice. Kindly remove one instance of it.

I really like this sample: `Is it possible for the bowling team to win the match if the batting team needs 1 run to win from 1 ball, a tie is not considered as a win`. It is a tricky one. If possible, can you add more situations like this one? More situations mean unique situations, not just increasing the number of balls. If it is not possible, that is also fine.

ai-scale commented:
@usama-openai 

I have removed the duplicate instance. Sorry for that.
I have also added couple of more situations.

usama-openai reviewed (approved):
This PR looks in good shape now. I'm approving this PR.

ai-scale commented:
@usama-openai @andrew-openai 
Thanks for merging my PR!
My email id was not public before, I have made it public now 
I'm really looking forward to experimenting with GPT4!

usama-openai commented:
You should see GPT-4 API access enabled in your account in the next few days.
